In [2]:
%pip install pandas

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 25.2 -> 25.3
[notice] To update, run: python.exe -m pip install --upgrade pip


In [4]:
import pandas as pd

In [ ]:
import numpy as np

In [10]:
import pandas as pd

df2 = pd.read_csv(
    r"C:\Users\varle\OneDrive\Documents\GitHub\Team8_PyMaster_PythonHackathon_December25\schema_2_ontario_final.csv"
)

df2.head()

week  fsa probable vulnerable is_most_recent fever_chills_shakes cough  \
0    15  M1P        n          n              y                   n     n   
1    16  N2P        n          y              y                   n     n   
2    16  N2P        n          y              n                   n     n   
3    16  M4P        n          n              y                   n     n   
4    16  L4N        n          n              y                   n     n   

  shortness_of_breath any_medical_conditions travel_outside_canada  \
0                   n                      y                     n   
1                   n                      y                     n   
2                   n                      y                     n   
3                   n                      y                     n   
4                   n                      y                     n   

  contact_with_illness   symptoms         conditions  ethnicity sex  \
0                    n        NaN                NaN        NaN   m   
1                    n  runnyNose  highBloodPressure      asian   m   
2                    n        NaN  highBloodPressure      asian   m   
3                    n        NaN                NaN  caucasian   f   
4                    n        NaN                NaN  caucasian   f   

              needs  age_1  
0               NaN  45-64  
1             other  45-64  
2             other  45-64  
3  emotionalSupport  26-44  
4  emotionalSupport  26-44

In [11]:
df2.head()
df2.tail()
df2.shape
df2.columns
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14932 entries, 0 to 14931
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   week                    14932 non-null  int64 
 1   fsa                     14932 non-null  object
 2   probable                14932 non-null  object
 3   vulnerable              14932 non-null  object
 4   is_most_recent          14932 non-null  object
 5   fever_chills_shakes     14932 non-null  object
 6   cough                   14932 non-null  object
 7   shortness_of_breath     14932 non-null  object
 8   any_medical_conditions  14932 non-null  object
 9   travel_outside_canada   14932 non-null  object
 10  contact_with_illness    14932 non-null  object
 11  symptoms                1796 non-null   object
 12  conditions              4728 non-null   object
 13  ethnicity               7101 non-null   object
 14  sex                     14932 non-null  object
 15  ne

Missing Value analysis

In [12]:
df2.isna().sum().sort_values(ascending=False)

symptoms                  13136
conditions                10204
ethnicity                  7831
needs                      7831
week                          0
fsa                           0
probable                      0
vulnerable                    0
is_most_recent                0
any_medical_conditions        0
shortness_of_breath           0
cough                         0
fever_chills_shakes           0
travel_outside_canada         0
contact_with_illness          0
sex                           0
age_1                         0
dtype: int64

Duplicate records


In [13]:
df1.duplicated().sum()

np.int64(3902)

3902 Duplicate rows are  duplicate in Data set

check Total rows

In [14]:
df1.shape

(14932, 17)

14932 -3902 =11030

2.Inspect duplicate rows

In [15]:
df2[df2.duplicated()].head()

week  fsa probable vulnerable is_most_recent fever_chills_shakes cough  \
45     15  K0H        n          y              n                   n     n   
71     15  M6G        n          n              n                   n     n   
101    15  M6S        n          n              y                   n     n   
128    15  K1R        n          n              n                   n     n   
129    15  K1R        n          n              n                   n     n   

    shortness_of_breath any_medical_conditions travel_outside_canada  \
45                    n                      y                     n   
71                    n                      y                     n   
101                   n                      y                     n   
128                   n                      y                     n   
129                   n                      y                     n   

    contact_with_illness symptoms conditions ethnicity sex needs  age_1  
45                     n      NaN        NaN       NaN   m   NaN    >65  
71                     n      NaN        NaN       NaN   m   NaN  45-64  
101                    n      NaN        NaN       NaN   f   NaN  45-64  
128                    n      NaN        NaN       NaN   f   NaN  45-64  
129                    n      NaN        NaN       NaN   f   NaN  45-64

drop Duplicate rows


In [16]:
df2_clean = df2.drop_duplicates()

 Verify duplicates are removed

In [17]:
df2_clean.duplicated().sum()

np.int64(0)

Data Cleaning Order
Reset index

In [18]:
df2_clean.reset_index(drop=True, inplace=True)

In [19]:
df2_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11030 entries, 0 to 11029
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   week                    11030 non-null  int64 
 1   fsa                     11030 non-null  object
 2   probable                11030 non-null  object
 3   vulnerable              11030 non-null  object
 4   is_most_recent          11030 non-null  object
 5   fever_chills_shakes     11030 non-null  object
 6   cough                   11030 non-null  object
 7   shortness_of_breath     11030 non-null  object
 8   any_medical_conditions  11030 non-null  object
 9   travel_outside_canada   11030 non-null  object
 10  contact_with_illness    11030 non-null  object
 11  symptoms                1633 non-null   object
 12  conditions              4216 non-null   object
 13  ethnicity               5980 non-null   object
 14  sex                     11030 non-null  object
 15  ne

In [21]:
print(df2_clean.columns.tolist())

['week', 'fsa', 'probable', 'vulnerable', 'is_most_recent', 'fever_chills_shakes', 'cough', 'shortness_of_breath', 'any_medical_conditions', 'travel_outside_canada', 'contact_with_illness', 'symptoms', 'conditions', 'ethnicity', 'sex', 'needs', 'age_1']


In [22]:
binary_cols = [
    'probable',
    'vulnerable',
    'is_most_recent',
    'fever_chills_shakes',
    'cough',
    'shortness_of_breath',
    'any_medical_conditions',
    'travel_outside_canada',
    'contact_with_illness'
]

In [24]:
binary_cols = [
    'probable',
    'vulnerable',
    'is_most_recent',
    'fever_chills_shakes',
    'cough',
    'shortness_of_breath',
    'any_medical_conditions',
    'travel_outside_canada',
    'contact_with_illness'
]

# Standardize yes/no values → 0/1
df2_clean[binary_cols] = (
    df2_clean[binary_cols]
    .apply(lambda col: col.astype(str).str.strip().str.lower())
    .replace({
        'y': 1, 'yes': 1, 'true': 1,
        'n': 0, 'no': 0, 'false': 0
    })
)

C:\Users\varle\AppData\Local\Temp\ipykernel_19664\3778471517.py:17: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  .replace({
C:\Users\varle\AppData\Local\Temp\ipykernel_19664\3778471517.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2_clean[binary_cols] = (


Check missing values

In [25]:
df2_clean.isna().sum().sort_values(ascending=False)

symptoms                  9397
conditions                6814
ethnicity                 5050
needs                     5050
week                         0
fsa                          0
probable                     0
vulnerable                   0
is_most_recent               0
any_medical_conditions       0
shortness_of_breath          0
cough                        0
fever_chills_shakes          0
travel_outside_canada        0
contact_with_illness         0
sex                          0
age_1                        0
dtype: int64

In [27]:
df2_clean['week'] = pd.to_numeric(df2_clean['week'], errors='coerce')

C:\Users\varle\AppData\Local\Temp\ipykernel_19664\1365960622.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2_clean['week'] = pd.to_numeric(df2_clean['week'], errors='coerce')


convert week in to  mont

In [30]:
df2['month'] = df2['date'].dt.to_period('M').astype(str)

KeyError: 'date'

Then convert week to numeric or datetime, not date.